core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -92,58 +85,94 @@ func (c *OutgoingConnectorHandler) HandleSingleNodeRequest(ctx context.Context,
 	// Create a subcontext with the timeout plus some margin for the gateway to process the request
 	timeoutDuration := time.Duration(req.TimeoutMs) * time.Millisecond
 	margin := 100 * time.Millisecond
-	ctx, cancel := context.WithTimeout(ctx, timeoutDuration+margin)
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, timeoutDuration+margin)
 	defer cancel()
 
-	payload, err := json.Marshal(req)
-	if err != nil {
-		return nil, fmt.Errorf("failed to marshal fetch request: %w", err)
-	}
+	sendRequest := c.createSendRequest(messageID, req)
 
-	ch, err := c.responses.new(messageID)
-	if err != nil {
-		return nil, fmt.Errorf("duplicate message received for ID: %s", messageID)
+	return c.handleSingleNodeRequest(ctxWithTimeout, sendRequest, uint(req.MaxRetries))
+}
+
+// handleSingleNodeRequest accepts a blocking function that returns an api message and error and either calls it directly or with
+// a retry strategy.
+func (c *OutgoingConnectorHandler) handleSingleNodeRequest(
+	ctx context.Context,
+	sendRequest func(context.Context) (*api.Message, error),
+	maxRetries uint,
+) (*api.Message, error) {
+	if maxRetries > 0 {
+		strategy := &retry.Strategy[*api.Message]{
+			MaxRetries: min(maxRetries, defaultMaxRetries),
+		}
+		return strategy.Do(ctx, c.lggr, sendRequest)
 	}
-	defer c.responses.cleanup(messageID)
 
-	lggr.Debugw("sending request to gateway")
+	return sendRequest(ctx)
```

Comment:
there is a request timeout, but by default we conservatively only make external calls once unless user specifies request for retry logic by passing a retry cap.  in all situations, context cancelation supersedes max retry count.